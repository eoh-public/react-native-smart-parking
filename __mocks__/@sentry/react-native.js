export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
