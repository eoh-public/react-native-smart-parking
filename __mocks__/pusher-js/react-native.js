const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
