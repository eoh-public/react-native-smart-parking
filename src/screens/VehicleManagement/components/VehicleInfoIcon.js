import React, { memo, useMemo } from 'react';
import { StyleSheet, View } from 'react-native';
import { IconFill } from '@ant-design/icons-react-native';
import { Icon } from '@ant-design/react-native';
import t from 'i18n';

import { Colors } from '../../../configs';
import Text from '../../../commons/Text';
import { CircleView } from '../../../commons';

const VehicleInfoIcon = memo(
  ({ style, car, showRemove, showDefault, onPressMinus }) => {
    const { is_default } = car;

    const isShowMissingIcon = useMemo(() => {
      const { name, plate_number, background } = car;
      const isMissingData = !(name && plate_number && background);
      return !(showRemove || showDefault) && isMissingData;
    }, [car, showDefault, showRemove]);

    return (
      <View style={[styles.container, style]}>
        {showRemove && (
          <Icon
            name={'minus'}
            color={Colors.Gray8}
            size={21}
            onPress={onPressMinus}
          />
        )}
        {!showRemove && is_default && (
          <View style={styles.default}>
            <Icon name={'check-circle'} color={Colors.Primary} size={21} />
            <Text style={styles.txtDefault}>{t('default')}</Text>
          </View>
        )}
        {showDefault && !is_default && (
          <CircleView size={21} borderWidth={1} borderColor={Colors.Gray5} />
        )}
        {isShowMissingIcon && (
          <IconFill
            name={'exclamation-circle'}
            color={Colors.Orange}
            size={21}
          />
        )}
      </View>
    );
  }
);

export default VehicleInfoIcon;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
  },
  default: {
    alignItems: 'center',
  },
  txtDefault: {
    fontSize: 10,
    lineHeight: 20,
    marginTop: 1.5,
  },
});
