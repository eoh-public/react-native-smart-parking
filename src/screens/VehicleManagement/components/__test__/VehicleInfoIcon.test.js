import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { StyleSheet } from 'react-native';
import VehicleInfoIcon from '../VehicleInfoIcon';

describe('Test VehicleInfoIcon', () => {
  const styles = StyleSheet.create({
    vehicleContainer: {
      position: 'absolute',
      top: 16,
      right: 16,
    },
  });
  const car = {
    name: 'Car',
    plate_number: '60B-999.99',
    background:
      'https://cdn-staging.eoh.io/image-6febeb51-2a4a-482e-94e4-38cd8e5e848e_jtutkUE.jpg',
  };
  let tree;

  test('render VehicleInfoIcon showRemove false', () => {
    const props = {
      car: { ...car, is_default: true },
      showRemove: false,
      showDefault: false,
      onPressMinus: () => {},
    };
    act(() => {
      tree = renderer.create(
        <VehicleInfoIcon style={styles.vehicleContainer} {...props} />
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render VehicleInfoIcon showRemove true', () => {
    const props = {
      car: { ...car, is_default: false },
      showRemove: true,
      showDefault: true,
      onPressMinus: () => {},
    };
    act(() => {
      tree = renderer.create(
        <VehicleInfoIcon style={styles.vehicleContainer} {...props} />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
