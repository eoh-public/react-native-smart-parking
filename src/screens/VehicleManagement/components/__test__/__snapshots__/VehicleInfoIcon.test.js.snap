// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test VehicleInfoIcon render VehicleInfoIcon showRemove false 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "position": "absolute",
        "right": 16,
        "top": 16,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#00979D",
            "fontFamily": "antoutline",
            "fontSize": 21,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": undefined,
            "lineHeight": undefined,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "fontSize": 10,
            "lineHeight": 20,
            "marginTop": 1.5,
          },
        ]
      }
    >
      MẶC ĐỊNH
    </Text>
  </View>
</View>
`;

exports[`Test VehicleInfoIcon render VehicleInfoIcon showRemove true 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
      Object {
        "position": "absolute",
        "right": 16,
        "top": 16,
      },
    ]
  }
>
  <Text
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#595959",
          "fontFamily": "antoutline",
          "fontSize": 21,
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        undefined,
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#D9D9D9",
          "borderRadius": 10.5,
          "borderWidth": 1,
          "height": 21,
          "width": 21,
        },
        undefined,
        undefined,
      ]
    }
  />
</View>
`;
